test/cli-validator/tests/thresholdValidator.js

```diff
@@ -0,0 +1,89 @@
+// the rule names are all snake case and need to stay that way. don't lint them
+/* eslint-disable camelcase */
+
+const intercept = require('intercept-stdout');
+const expect = require('expect');
+const stripAnsiFrom = require('strip-ansi');
+
+const commandLineValidator = require('../../../src/cli-validator/runValidator');
+
+describe('test the .thresholdrc limits', function() {
+  it('should show error and set exit code to 1 when warning limit exceeded', async function() {
+    const capturedText = [];
+
+    const program = {};
+    program.args = ['./test/cli-validator/mockFiles/circularRefs.yml'];
+    program.limits = './test/cli-validator/mockFiles/thresholds/.fiveWarnings';
+    program.default_mode = true;
+
+    const unhookIntercept = intercept(function(txt) {
+      capturedText.push(stripAnsiFrom(txt));
+      return '';
+    });
+
+    const exitCode = await commandLineValidator(program);
+
+    unhookIntercept();
+
+    expect(exitCode).toEqual(1);
+
+    expect(capturedText[capturedText.length - 1].slice(0, 18)).toEqual(
+      `Number of warnings`
+    );
+  });
+
+  it('should print errors for unsupported limit options and invalid limit values', async function() {
+    const capturedText = [];
+
+    const program = {};
+    program.args = ['./test/cli-validator/mockFiles/clean.yml'];
+    program.limits = './test/cli-validator/mockFiles/thresholds/.invalidValues';
+    program.default_mode = true;
+
+    const unhookIntercept = intercept(function(txt) {
+      capturedText.push(stripAnsiFrom(txt));
+      return '';
+    });
+
+    const exitCode = await commandLineValidator(program);
+
+    unhookIntercept();
+
+    // limit values invalid, so default limit, Number.MAX_VALUE, used
+    expect(exitCode).toEqual(0);
+
+    const allOutput = capturedText.join('');
+
+    expect(
+      allOutput.includes('"population" limit not supported.') &&
+        allOutput.includes('Value provided for warnings')
+    ).toEqual(true);
+  });
+
+  it('should give exit code 0 when warnings limit not exceeded', async function() {
+    const program = {};
+    program.args = ['./test/cli-validator/mockFiles/clean.yml'];
+    program.limits = './test/cli-validator/mockFiles/thresholds/.zeroWarnings';
+    program.default_mode = true;
+
+    const exitCode = await commandLineValidator(program);
+
+    expect(exitCode).toEqual(0);
+  });
+
+  it('should give an error for invalid JSON', async function() {
+    const program = {};
+    program.args = ['./test/cli-validator/mockFiles/clean.yml'];
+    program.limits = './test/cli-validator/mockFiles/thresholds/.invalidJSON';
+    program.default_mode = true;
+
+    try {
+      await commandLineValidator(program);
+      // error should be thrown for invalid JSON
+      expect(true).toEqual(false);
+    } catch (err) {
+      // invalid JSON in .thresholdrc, reject with exit code 2
+      expect(true).toEqual(true);
```

Comment:
What??

These look like they were leftover from debugging. These should assert what you're actually testing for

Reply:
I am expecting an error to be thrown while trying to parse the invalid JSON in the .invalidJSON file. I just wanted the catch block to execute for this test.

Reply:
Oh okay, I understand. This is a crafty way doing that but there is [an official way](https://jestjs.io/docs/en/expect#tothrowerror) (that is more readable) 🙂

Reply:
Oh nice! Thank you.